src/transform/src/join_implementation.rs

```diff
@@ -941,9 +955,16 @@ fn install_lifted_mfp(
                     #[allow(deprecated)]
                     expr.visit_mut_pre_post(
                         &mut |e| {
-                            if let MirScalarExpr::Column(c) = e {
+                            // This has to be a loop! This is because it can happen that the new
+                            // expression is again a column reference, in which case the visitor
+                            // wouldn't be called on it again. (The visitation continues with the
+                            // children of the new expression, but won't visit the new expression
+                            // itself again.)
+                            while let MirScalarExpr::Column(c) = e {
```

Comment:
Could we instead canonicalize the map before we visit `expr`? Not a huge concern, but it would avoid the loop here.